Make drag handle visible on hover in nested

jasmussen opened the pull request:
In nested contexts, the drag handle doesn't have a background on hover like the up/down arrows. This PR refactors things a bit and fixes it up.

Before:

<img width="272" alt="screen shot 2018-09-20 at 11 43 20" src="https://user-images.githubusercontent.com/1204802/45811579-bbcea600-bccd-11e8-8ab0-a357f7c9435c.png">

After:

<img width="304" alt="screen shot 2018-09-20 at 12 05 25" src="https://user-images.githubusercontent.com/1204802/45811583-bec99680-bccd-11e8-8f76-5302aec6ac3a.png">

Commit:
Make drag handle visible on hover in nested

In nested contexts, the drag handle doesn't have a background on hover like the up/down arrows. This PR refactors things a bit and fixes it up.

Commit:
Forgot one thing.

kjellr commented:
Thanks, Joen — this is looking good, but I wonder if we can adjust the spacing a little bit before merging? Right now the top of the drag handle's top border overlaps with the bottom border of the up arrow, and the bottom border bumps right up against the top of the down arrow. It looks a little imbalanced: 

<img width="120" alt="screen shot 2018-09-20 at 10 52 46 am" src="https://user-images.githubusercontent.com/1202812/45827042-ab193280-bcc3-11e8-977e-26863c8f5788.png">

It'd be great if there was a 1px border in between all: 

![1px](https://user-images.githubusercontent.com/1202812/45827246-077c5200-bcc4-11e8-87f4-ec24bea37d84.png)

chrisvanpatten commented:
It frustrates me a little bit that we want to target nested blocks separately. I mean I understand _why_, I just feel like there has to be a middle road that works in both contexts and feels a little less kludgy.

One concept is something like this that treats the entire mover as single area, like so:

![edit_post_ _tomodomo_ _wordpress](https://user-images.githubusercontent.com/1231306/45854307-7edbd100-bd18-11e8-943d-0b7e29dfd3b2.png)

(Note that in the screenshot I did also align the top edge of the movers and the block canvas. I have a lot of reasons why I think that's right, which I spent some time typing up and then decided not to share them so as not to derail my main point about nested/non-nested consistency. So ignore the vertical positioning, and just pay attention to the single border rather than individual button borders. 😄)

Commit:
Address feedback.

jasmussen commented:
Great catches. I addressed the feedback:

<img width="261" alt="screen shot 2018-09-21 at 10 07 47" src="https://user-images.githubusercontent.com/1204802/45868574-686c5e80-bd86-11e8-806b-d5381a4f903c.png">

The borders are stacked rather than space between, I have a slight preference for that.

It also works when you multiselect in a nested context:

<img width="204" alt="screen shot 2018-09-21 at 10 08 41" src="https://user-images.githubusercontent.com/1204802/45868608-84700000-bd86-11e8-87d7-aae7d7502ffe.png">

The meta issue here, as beautifully noted and addressed by @chrisvanpatten, is that this isn't ideal. I totally agree, but thankfully I feel we've made strides by moving the ellipsis to the block toolbar and simplified things like the drag handle, at least it's getting there.

I also happen to prefer the mockup Chris provided, in fact I tried doing something like that, but the current setup doesn't easily lend itself to this design. That doesn't mean it shouldn't be done, just that I couldn't do it fast, and I'd rather have small iterative improvements in the mean time.

Commit:
Remove debug style.

chrisvanpatten commented:
@jasmussen agreed wholeheartedly. I have a WIP branch with several other tweaks that I can push up but that shouldn’t block this. I say :ship: it!

kjellr reviewed (approved):
🚢 Thanks, @jasmussen! 

I also like that exploration by @chrisvanpatten — it'd be great to see something like that happen in the future.
